### PR TITLE
Replaces clockwork component initials with icons

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
imageadd: Clockwork slabs now show icons of the components used in scripture instead of initials.
/:cl:

Thank you to @ChangelingRain for the sprites, you rock!
Also thank you @RemieRichards for helping me get image code that didn't suck, you also rock!

I saw this suggested and thought that it may be a good idea, and quicker to grasp for new players than the abbreviations that used to exist. Instead of using abbreviations for components, the slab now uses small 16x16 icons of each component.

### CODE REVIEW

**Comment by RemieRichards** [comment]
>thank joan for sprites
>don't thank me for walking you through getting icons into tgui.

smh tbh.

**Comment by Xhuis** [comment]
I forgot, okay! There.

File: code/game/gamemodes/clock_cult/clock_items/clockwork_slab.dm [hunk 1]
```diff
@@ -570,14 +570,14 @@
 	data["components"] = stored_components.Copy()
 	var/list/temp_data = list("<font color=#B18B25>")
 	for(var/i in data["components"]) //display the slab's components
-		temp_data += "<font color=[get_component_color_bright(i)]>[get_component_acronym(i)] <b>[data["components"][i]]</b></font>"
+		temp_data += "<font color=[get_component_color_bright(i)]>[get_component_icon(i)] <b>[data["components"][i]]</b></font>"
 		if(i != HIEROPHANT_ANSIBLE)
 			temp_data += " "
 		else
-			temp_data += " ("
+			temp_data += " <br>("
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_items/clockwork_slab.dm]: Remove this <br> or you'll have to make the slab interface taller

Comment (Xhuis) [inline code/game/gamemodes/clock_cult/clock_items/clockwork_slab.dm]: It's to separate the component count and the "with global cache" count; it doesn't seem to have any adverse effects, as you can see it in the screenshot. Is there something I missed that I should be accounting for?

Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_items/clockwork_slab.dm]: Components (with Global Cache) isn't separated, and adding a `<br>` there will make the interface taller. Accordingly, remove this one so you don't have to adjust the interface's size.

Comment (Xhuis) [inline code/game/gamemodes/clock_cult/clock_items/clockwork_slab.dm]: I still think it's weeeeird like that, but you also make a point. OP is what it looks like with the break removed.

**Comment by ChangelingRain** [comment]
Uuuugh, it still adds a little bit of height you can see the clockwork obelisk getting cut off at the bottom there

**Comment by Xhuis** [comment]
How big of a worry is it? I'm not sure if it was symmetrical where it had just the initials.

**Review by Cyberboss — APPROVED** [review]

**Comment by ChangelingRain** [comment]
With #29173, I have a solution: http://puu.sh/wGdOG/5c5695074c.dmi

File: code/game/gamemodes/clock_cult/clock_helpers/component_helpers.dm [hunk 2]
```diff
@@ -62,6 +62,10 @@
 		else
 			return null
 
+//returns a component icon from a component id
+/proc/get_component_icon(id)
+	return "<img style='width:16px; height:16px' src='data:image/png;base64,[icon2base64(icon('icons/obj/tgui_components.dmi', id))]'/>"
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_helpers/component_helpers.dm]: I feel like you should be cacheing this for some reason but I dunno, @RemieRichards?

**Comment by Xhuis** [comment]
Even if that new sprite PR doesn't get in (let's be honest, we both know it's gonna get in) I'll use the new ones, as they're much easier on the eyes for just this purpose anyway.

**Comment by Xhuis** [comment]
New cogwheel sprites are in. Thanks again!

**Comment by ChangelingRain** [comment]
Hmm, try these, they're slightly smaller since I removed part of the eye and ansible sprites that weren't really visible http://puu.sh/wGlEb/ad16866bd7.dmi

**Comment by Xhuis** [comment]
I don't think they're stretching the buttons at all. As far as I can tell they're the same size as the text.
Will try these out once I finish up what I'm working on.

**Comment by ChangelingRain** [comment]
I know, for a fact, that they are stretching the buttons, because the buttons with icons are larger than the buttonless quickbind icons.

**Comment by Xhuis** [comment]
Hrm. Do you know exactly how tall the max is before it starts stretching? If 16x16 is pushing it like this then I feel like 15x15 or something should be fine.

**Comment by ChangelingRain** [comment]
I'm not sure what the max is but try smaller numbers

**Review by ChangelingRain — APPROVED** [review]

**Pull request merged**